src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowPathsSpec.groovy

```diff
@@ -38,4 +35,30 @@ class HaFlowPathsSpec extends HealthCheckSpecification {
         def actualException = thrown(HttpClientErrorException)
         new HaFlowNotFoundExpectedError(~/HA-flow ${flowId} not found\./).matches(actualException)
     }
+
+    @Tidy
+    def "HA flow main path is not overlapped with protected path"() {
+        assumeTrue(useMultitable, "HA-flow operations require multiTable switch mode")
+        given: "An HA-flow with protected path"
+        def swT = topologyHelper.findSwitchTripletForHaFlowWithProtectedPaths()
+        assumeTrue(swT != null, "These cases cannot be covered on given topology:")
+
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        haFlowRequest.allocateProtectedPath = true
+        HaFlowExtended haFlow = swT.addHaFlow(haFlowRequest)
```

Comment:
I'd recommend to hide this creation into SwitchTriplet class. Keeping in mind, that random Ha-Flow won't work here, I'd recommend to use builder. I.e. code in spec should look like this:
```
HaFlowExtended haFlow = swT.customHaFlow() //this is static method that returns builder
                                                        .withProtectedPath(true)
                                                        .with....() //for now we don't need anything else
                                                        .create()
```

For easier cases, `swt.createRandomHaFlow()` would be good.
You can either call HaFlowHelper's methods from SwitchTriplet, or copy-paste them, or rewrite yourself, whatever suits you more at the moment.

Reply:
The new logic has been added. Could you please check?